text/0000-tracked-properties.md

```diff
@@ -227,34 +248,350 @@ memoization) if a property shows up as a bottleneck during profiling.
+computed property is accessed.
+
+In the edge case of non-computed, non-tracked property access, this is not the
+case. The property may be completely undecorated, meaning it may never have had
+a tag in the first place. With standard Javascript property access, we won't
+have an opportunity to _add_ a tag either. Autotrack will not work in these
+cases:
+
+```js
+const Config = Service.extend({
+  polling: {
+    shouldPoll: false,
+    pollInterval: -1,
+  },
+
+  init() {
+    this._super(...arguments);
+
+    fetch('config/api/url')
+      .then(r => r.json())
+      .then(polling => set(this, 'polling', polling));
+  }
+})
+
+class SomeComponent extends Component {
+  @service config;
+
+  @tracked
+  get pollInterval() {
+    // When we access shouldPoll and pollInterval, they don't have tags or
+    // wrapped getters. We have no idea that they could update in the future.
+    let { shouldPoll, pollInterval } = this.config.polling;
+
+    return shouldPoll ? pollInterval : -1;
+  }
+}
+```
+
+To support this case, `Ember.get` will manually opt in to adding the tags for
+any properties which are accessed during a tracked getter. This allows tracked
+getters to safely access services and objects which are still using the legacy
+change tracking system:
+
+```js
+const Config = Service.extend({
+  polling: {
+    shouldPoll: false,
+    pollInterval: -1,
+  },
+
+  init() {
+    this._super(...arguments);
+
+    fetch('config/api/url')
+      .then(r => r.json())
+      .then(polling => set(this, 'polling', polling));
+  }
+})
+
+class SomeComponent extends Component {
+  @service config;
+
+  @tracked
+  get pollInterval() {
+    // When we access shouldPoll and pollInterval, they don't have tags or
+    // wrapped getters. We have no idea that they could update in the future.
+    let shouldPoll = this.get('config.polling.shouldPoll');
+    let pollInterval = this.get('config.polling.pollInterval');
+
+    return shouldPoll ? pollInterval : -1;
+  }
+}
+```
+
+This solution will _only_ be necessary in cases where users do not know ahead of
+time whether a property is A. immutable, B. tracked, or C. a computed property.
+As time goes on and more and more services, addons, and apps are converted to
+`@tracked`, the need for `Ember.get` will become smaller and smaller, until it
+can be removed from the framework entirely (along with `Ember.set`).
+
+### Computed Property Interop with Tracked
+
+The previous section covers the cases where classic Ember properties are
+accessed from tracked properties, but we also have to define the behavior for
+when the opposite occurs: tracked properties are accessed by from a classic
+context.
+
+There are two major ways this can occur:
+
+1. Computed Properties
+2. Observers
+
+Observers are strictly within the old paradigm, so tracked properties will _not_
+attempt to interoperate with them. Using `Ember.set` on a tracked property will
```

Comment:
Kind of a nitpick, but using the `Ember.set` notation is kind of strange given that we're moving people away from the `Ember` global. Should maybe define this early in the document as "Ember's `set()` function (exported from the `@ember/object` module)", then just use `set()` rather than `Ember.set()` throughout.

Reply:
Yeah I wish there was an easy standard way to inline refer to a module and export. Will update.